### PR TITLE
Feature/contract agreements master refactoring

### PR DESCRIPTION
**Pull request by vonElfvin**

### CODE REVIEW

File: src/main/java/com/hedvig/productPricing/service/serviceIntegration/customerio/CustomerIO.kt [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.hedvig.productPricing.service.serviceIntegration.customerio
+
+import com.hedvig.productPricing.service.web.dto.InsuranceCompaniesSE
+
+interface CustomerIO {
```
Comment (meletios) [inline src/main/java/com/hedvig/productPricing/service/serviceIntegration/customerio/CustomerIO.kt]: Is this related? 🤔

Comment (vonElfvin) [reply src/main/java/com/hedvig/productPricing/service/serviceIntegration/customerio/CustomerIO.kt]: No and I have no idea how it got in here...

File: pom.xml [hunk 2]
```diff
@@ -359,6 +359,11 @@
       <artifactId>json-path-assert</artifactId>
       <version>2.2.0</version>
     </dependency>
+    <dependency>
+      <groupId>com.neovisionaries</groupId>
+      <artifactId>nv-i18n</artifactId>
+      <version>1.27</version>
+    </dependency>
```
Comment (vonElfvin) [inline pom.xml]: Do we want to use an enum for country, then I really like this one. It's really easy to use and we don't have to maintain it ourselves.

File: src/main/java/com/hedvig/contract/apartments/aggregates/ApartmentAgreement.kt [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+import java.time.Instant
+import java.time.LocalDate
+import java.util.UUID
+import javax.money.MonetaryAmount
+
+data class ApartmentAgreement(
+  override val id: UUID,
+  override val basePremium: MonetaryAmount,
+  override val lineOfBusiness: ApartmentLineOfBusiness,
+  override val termsAndConditions: String, // FIXME: Please determine what this should be
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/apartments/aggregates/ApartmentAgreement.kt]: We need to talk to Tomas about this one I guess

File: src/main/java/com/hedvig/contract/apartments/aggregates/ApartmentAgreement.kt [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    lineOfBusiness: LineOfBusinessStereotype?,
+    termsAndConditions: String,
+    certificate: InsuranceCertificatePdf?
+  ): TAgreement = copy(
+    id = id,
+    from = from,
+    to = to,
+    basePremium = premium,
+    lineOfBusiness = lineOfBusiness as ApartmentLineOfBusiness,
+    termsAndConditions = termsAndConditions,
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/apartments/aggregates/ApartmentAgreement.kt]: These should probably not be included in the copy. TODO: Remove from the BaseAgreement

File: src/main/java/com/hedvig/contract/apartments/models/ApartmentAgreementState.kt [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+  to: LocalDate?,
+  premium: BigDecimal,
+  currency: String,
+  termsAndConditions: String,
+  certificate: InsuranceCertificatePdf?,
+  @Enumerated(EnumType.STRING)
+  override val lineOfBusiness: ApartmentLineOfBusiness,
+  @Embedded
+  val address: Address,
+  @ElementCollection
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/apartments/models/ApartmentAgreementState.kt]: Use @OneToMany instead? Or maybe even @ManyToMany?

File: src/main/java/com/hedvig/contract/apartments/models/ApartmentAgreementState.kt [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+  val coInsured: MutableList<CoInsured>,
+  val squareMeters: Long,
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/apartments/models/ApartmentAgreementState.kt]: Good thing we use a Long for when we are gonna insure 32768 SQM apartments 😄

File: src/main/java/com/hedvig/contract/apartments/aggregates/ApartmentAgreement.kt [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+        id = compatAgreement.id,
+        basePremium = Money.of(compatAgreement.basePremium!!, CompatAgreement.SEK),
+        lineOfBusiness = ApartmentLineOfBusiness.valueOf(compatAgreement.lineOfBusiness!!),
+        from = compatAgreement.from,
+        to = compatAgreement.to,
+        address = Address(
+          compatAgreement.addressStreet,
+          compatAgreement.addressCoLine,
+          compatAgreement.addressPostalCode,
+          compatAgreement.addressCity,
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/apartments/aggregates/ApartmentAgreement.kt]: This is an easy location to swap city and postal code if the city is the postal code!

Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/apartments/aggregates/ApartmentAgreement.kt]: Also, let's strip the zipCode from any spaces (i.e. make 163 63 to 16363) for consistency. Trimming spaces at the end would also be nice.

File: src/main/java/com/hedvig/contract/houses/models/HouseAgreementState.kt [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  to: LocalDate?,
+  premium: BigDecimal,
+  currency: String,
+  termsAndConditions: String,
+  certificate: InsuranceCertificatePdf?,
+  override val lineOfBusiness: HouseLineOfBusiness? = null,
+  @Embedded
+  val address: Address,
+  @ElementCollection
+  val coInsured: List<CoInsured>,
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/houses/models/HouseAgreementState.kt]: @OneToMany? @ManyToMany?

Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/houses/models/HouseAgreementState.kt]: Would actually be cool if "CoInsured" had a @ManyToMany" with what contracts he/she is coinsured at.

Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/houses/models/HouseAgreementState.kt]: Also, is the holderMemberId included in this list??

File: src/main/java/com/hedvig/contract/shared/aggregates/BaseAgreement.kt [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  override val to: LocalDate?,
+  override val certificate: InsuranceCertificatePdf?
+) : AgreementStereotype {
+  abstract fun <T : AgreementStereotype> copy_(
+    id: UUID = this.id,
+    from: LocalDate? = this.from,
+    to: LocalDate? = this.to,
+    premium: MonetaryAmount = this.basePremium,
+    lineOfBusiness: LineOfBusinessStereotype? = this.lineOfBusiness,
+    termsAndConditions: String = this.termsAndConditions,
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/aggregates/BaseAgreement.kt]: Note to self: Remove these

File: src/main/java/com/hedvig/contract/shared/models/AgreementState.kt [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+  private val premium: BigDecimal,
+  private val currency: String,
+
+  override val termsAndConditions: String,
+
+  @Embedded
+  override var certificate: InsuranceCertificatePdf?,
+
+  val createdAt: Instant,
+  var removedAt: Instant? = null
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/models/AgreementState.kt]: One cool thing would be to add "replacedBy" so we can show "replacement history" as well for an agreement

File: src/main/java/com/hedvig/contract/shared/models/ContractState.kt [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+import java.time.Instant
+import java.time.LocalDate
+import java.util.UUID
+import javax.persistence.Entity
+import javax.persistence.Id
+import javax.persistence.Inheritance
+import javax.persistence.InheritanceType
+
+@Entity
+@Inheritance(strategy = InheritanceType.TABLE_PER_CLASS)
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/models/ContractState.kt]: Explore using@MappedSuperclass instead, they seem very similar

File: src/main/java/com/hedvig/contract/shared/models/ContractState.kt [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+abstract class ContractState<TAgreement : AgreementState>(
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/models/ContractState.kt]: Maybe add a "replaceAgreement" so you don't have to do that 2 step process (i.e. remove then add) outside

File: src/main/java/com/hedvig/contract/shared/models/ContractStateFactory.kt [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+class ContractStateFactory {
+  @Suppress("TYPE_MISMATCH")
+  private fun createEmptyContractState(
+    contractId: UUID,
+    holderMemberId: String,
+    agreement: BaseAgreement,
+    switchedFrom: String?,
+    createdAt: Instant
+  ): ContractState<AgreementState> {
+    return when (agreement) {
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/models/ContractStateFactory.kt]: Test using object : ContractState<AgreementState> here

File: src/main/java/com/hedvig/contract/apartments/models/ApartmentLineOfBusiness.kt [hunk 14]
```diff
@@ -0,0 +1,10 @@
+package com.hedvig.contract.apartments.models
+
+import com.hedvig.contract.shared.stereotypes.LineOfBusinessStereotype
+
+enum class ApartmentLineOfBusiness : LineOfBusinessStereotype {
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/apartments/models/ApartmentLineOfBusiness.kt]: I think that LineOfBusinessStereotype should be empty and not include

```
HOUSEHOLD
HOUSEHOLD_STUDENT
HOUSEHOLD_BUILDING
```
Since we dont care so much inside the agreement about those types eller?

File: src/main/java/com/hedvig/contract/shared/models/ContractStateFactory.kt [hunk 15]
```diff
@@ -0,0 +1,74 @@
...
+    }
+  }
+
+  private fun createAgreementState(agreement: BaseAgreement, createdAt: Instant): AgreementState {
+    return when (agreement) {
+      is ApartmentAgreement -> agreement.toAgreementState(createdAt)
+      is HouseAgreement -> agreement.toAgreementState(createdAt)
+      else -> throw Exception("Unable to create agreement state")
+    }
+  }
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/models/ContractStateFactory.kt]: This method is redundant really

File: src/main/java/com/hedvig/contract/shared/models/objects/CoInsured.kt [hunk 16]
```diff
@@ -0,0 +1,9 @@
+package com.hedvig.contract.shared.models.objects
+
+import javax.persistence.Embeddable
+
+@Embeddable
+data class CoInsured(
+  val memberId: String? = null
+  // val ssn: String? = null FIXME: How to add coInsured that is not member?
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/models/objects/CoInsured.kt]: Add a @ManyToMany contracts to this one?

File: src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt [hunk 17]
```diff
@@ -0,0 +1,306 @@
...
+
+  @EventHandler
+  fun on(
+    event: ContractWithPendingAgreementCreated<BaseAgreement>,
+    @Timestamp timestamp: Instant,
+    @MetaDataValue(MetadataKeys.ORIGINAL_TIMESTAMP) originalTimeStamp: Instant?
+  ) {
+    logger.info("[${event.javaClass.simpleName}] Handling event for [memberId=${event.holderMemberId}]")
+
+    val contractState = contractStateFactory.createContractStateWithPendingAgreement(
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt]: May as well send the event

File: src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt [hunk 18]
```diff
@@ -0,0 +1,306 @@
...
+
+  @EventHandler
+  fun on(
+    event: ContractWithActiveAgreementCreated<BaseAgreement>,
+    @Timestamp timestamp: Instant,
+    @MetaDataValue(MetadataKeys.ORIGINAL_TIMESTAMP) originalTimeStamp: Instant?
+  ) {
+    logger.info("[${event.javaClass.simpleName}] Handling event for [memberId=${event.holderMemberId}]")
+
+    val contractState = contractStateFactory.createContractStateWithActiveAgreement(
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt]: may as well send the event instead

File: src/main/java/com/hedvig/contract/houses/models/HouseLineOfBusiness.kt [hunk 19]
```diff
@@ -0,0 +1,5 @@
+package com.hedvig.contract.houses.models
+
+import com.hedvig.contract.shared.stereotypes.LineOfBusinessStereotype
+
+enum class HouseLineOfBusiness : LineOfBusinessStereotype
```
Comment (meletios) [inline src/main/java/com/hedvig/contract/houses/models/HouseLineOfBusiness.kt]: Same applies to here or?

File: src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt [hunk 20]
```diff
@@ -0,0 +1,306 @@
...
+    @MetaDataValue(MetadataKeys.ORIGINAL_TIMESTAMP) originalTimeStamp: Instant?
+  ) {
+    val contractState = findContractOrThrow(event.contractId)
+
+    logger.info("[${event.javaClass.simpleName}] Handling event for [memberId=${contractState.holderMemberId}]")
+
+    val agreementToActive = contractState.pendingAgreement!!
+    agreementToActive.from = event.from
+    contractState.addAgreement(agreementToActive)
+    contractState.pendingAgreement = null
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt]: Make into a method on the ContractState instead

File: src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt [hunk 21]
```diff
@@ -0,0 +1,306 @@
...
+    logger.info("[${event.javaClass.simpleName}] Handling event for [memberId=${contractState.holderMemberId}]")
+
+    contractState.findAgreementOrThrow(event.previousAgreementId).removedAt = originalTimeStamp ?: timestamp
+    val agreementToReplaceWith = event.newAgreement.toAgreementState(originalTimeStamp ?: timestamp)
+
+    if (contractState.agreements?.removeIf { agreementState -> agreementState.id == agreementToReplaceWith.id } == true) {
+      logger.info("[${event.javaClass.simpleName}] Agreement Removed [agreementId=${agreementToReplaceWith.id}][memberId=${contractState.holderMemberId}]")
+    }
+
+    contractState.addAgreement(agreementToReplaceWith)
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt]: Looks to me like we remove and add the same agreement here right?

File: src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt [hunk 22]
```diff
@@ -0,0 +1,306 @@
...
+  }
+
+  @EventHandler
+  fun on(event: ContractTerminated) {
+    val contractState = findContractOrThrow(event.contractId)
+
+    logger.info("[${event.javaClass.simpleName}] Handling event for [memberId=${contractState.holderMemberId}]")
+
+    contractState.findAgreementOrThrow(event.agreementToTerminateId).to = event.terminationDate
+    contractState.terminationDate = event.terminationDate
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt]: Add as method on contractstate as well?

File: src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt [hunk 23]
```diff
@@ -0,0 +1,306 @@
...
+import org.axonframework.messaging.annotation.MetaDataValue
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+import java.time.Instant
+import java.util.UUID
+
+@Component
+@Transactional
+@ProcessingGroup(ContractStateProjection.CONTRACT_STATE)
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt]: Use its own ProcessingGroup or not?

File: src/main/java/com/hedvig/contract/shared/stereotypes/ContractStereotype.kt [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+    get() = pendingAgreement != null
+
+  val isTerminated: Boolean
+    get() = terminationDate != null
+
+  val isValid: Boolean
+    get() = when (hasPendingAgreement) {
+      true -> agreementsTimeline.isEmpty()
+      false -> agreementsTimeline.isValid(isTerminated)
+    }
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/stereotypes/ContractStereotype.kt]: This one can be extended A LOT if we want, maybe that would be nice? I.e. checking the validity of the pending agreement if present, checking so the terminationDate and the timeline ends on the same date etc.

Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/shared/stereotypes/ContractStereotype.kt]: It can be used to query out all invalid timelines for example

File: src/main/java/com/hedvig/contract/shared/timeline/Timeline.kt [hunk 25]
```diff
@@ -0,0 +1,164 @@
...
+
+  operator fun get(id: UUID): T = first { thing -> thing.id == id }
+
+  fun has(id: UUID): Boolean = find { thing -> thing.id == id } != null
+
+  fun getPartsThatBeginAfter(date: LocalDate): List<T> {
+    return this.filter { part -> part.partBeginning.isAfter(date) }
+  }
+
+  fun get(from: LocalDate, to: LocalDate?): List<T> {
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/timeline/Timeline.kt]: Add more explicit name here for what it does maybe

File: src/main/java/com/hedvig/contract/shared/timeline/Timeline.kt [hunk 26]
```diff
@@ -0,0 +1,164 @@
...
+    if (!this.isTerminated) {
+      throw TimelineNotTerminatedException.notTerminated()
+    }
+  }
+
+  private fun ensureActive() {
+    if (this.isTerminated) {
+      throw TimelineTerminatedException.terminated()
+    }
+  }
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/shared/timeline/Timeline.kt]: These exceptions can probably be more explicit with includoing the timeline id, what parts are affected etc. For faster fixing in the future.

File: src/main/java/com/hedvig/contract/translators/ProductToContractEventListener.kt [hunk 27]
```diff
@@ -138,23 +142,66 @@ class ProductToContractEventListener(
...
       val latestCompatAgreement = getLatestCompatAgreement(e.memberId, timeOfKnowledge)
 
       if (latestCompatAgreement != null && !latestCompatAgreement.skipAgreement) {
+        if (latestCompatAgreement.replacePreviousAgreement) {
+          try {
+            commandBus.sendAndWait<Void>(
+              ReplaceAgreement(
+                contractId = contractId,
+                oldAgreementId = latestCompatAgreement.replacingProductId!!,
+                newAgreement = ApartmentAgreement.fromCompatAgreement(latestCompatAgreement),
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/translators/ProductToContractEventListener.kt]: Don't forget house! It's as easy as:

```suggestion
                newAgreement = when(latestCompatAgreement.type) {
                HOUSE -> HouseAgreement.fromCompatAgreement(latestCompatAgreement)
                else -> ApartmentAgreement.fromCompatAgreement(latestCompatAgreement)
              
```

File: src/main/java/com/hedvig/contract/validator/ContractProductValidator.kt [hunk 28]
```diff
@@ -1,26 +1,30 @@
...
 import java.time.Instant
 import java.time.LocalDate
-import org.springframework.stereotype.Component
 
 @Component
 class ContractProductValidator(
   private val productRepository: ProductRepository,
   private val insuranceService: InsuranceService,
-  private val apartmentContractStateRepository: ApartmentContractStateRepository,
+  private val contractStateRepository: ContractStateRepository<ApartmentAgreementState, ContractState<ApartmentAgreementState>>,
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/validator/ContractProductValidator.kt]: Not house?

File: src/main/java/com/hedvig/contract/validator/ContractProductValidator.kt [hunk 29]
```diff
@@ -59,12 +69,13 @@ class ContractProductValidator(
 
   private fun findDifferences(
     product: ProductEntity,
-    contract: ApartmentContractState
+    inputContract: ContractState<ApartmentAgreementState>
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/validator/ContractProductValidator.kt]: No house?

File: src/main/java/com/hedvig/contract/web/TestController.kt [hunk 30]
```diff
@@ -0,0 +1,100 @@
...
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import java.math.BigDecimal
+import java.time.Instant
+import java.time.LocalDate
+import java.util.UUID
+
+@RestController
+@RequestMapping("/_/test")
+class TestController constructor(
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/contract/web/TestController.kt]: TODO: Delete

File: src/main/java/com/hedvig/productPricing/service/configuration/AxonConfiguration.java [hunk 31]
```diff
@@ -86,6 +86,12 @@ public void configure(EventProcessingConfiguration config) {
       x ->
         TrackingEventProcessorConfiguration.forSingleThreadedProcessing()
           .andInitialTrackingToken(StreamableMessageSource::createHeadToken));
+
+    config.registerTrackingEventProcessor(
+      "CustomerIO",
+      x ->
+        TrackingEventProcessorConfiguration.forSingleThreadedProcessing()
+          .andInitialTrackingToken(StreamableMessageSource::createHeadToken));
```
Comment (vonElfvin) [inline src/main/java/com/hedvig/productPricing/service/configuration/AxonConfiguration.java]: Goddamit, remove this too :D

File: src/test/java/com/hedvig/contract/shared/models/ContractRepositoryTest.kt [hunk 32]
```diff
@@ -0,0 +1,134 @@
...
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner
+import java.math.BigDecimal
+import java.time.Instant
+import java.time.LocalDate
+import java.util.UUID
+
+@DataJpaTest
+@RunWith(SpringJUnit4ClassRunner::class)
+class ContractRepositoryTest {
```
Comment (vonElfvin) [inline src/test/java/com/hedvig/contract/shared/models/ContractRepositoryTest.kt]: I would like to test how it behaves when you "findByHolderMemberId" and that member has both house and apartment.

File: pom.xml (hunk 2 above)
Comment (meletios) [reply pom.xml]: Nice, will replace our own enum then

File: src/main/java/com/hedvig/contract/validator/ContractProductValidator.kt (hunk 28 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/validator/ContractProductValidator.kt]: Lets not care for houses. We can check them manually 😁

File: src/main/java/com/hedvig/contract/validator/ContractProductValidator.kt (hunk 29 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/validator/ContractProductValidator.kt]: Lets not care for houses. We can check them manually 😁

File: src/main/java/com/hedvig/contract/houses/models/HouseAgreementState.kt (hunk 8 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/houses/models/HouseAgreementState.kt]: we have the memberId of the CoInsured person

File: src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt (hunk 21 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt]: Yes this is because some of the events (especially the modifcations) are 2-3-5 times for the same aggregate and it basically removing eachother couple of times

File: src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt (hunk 23 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt]: Yes its better if its own.

Comment (meletios) [reply src/main/java/com/hedvig/contract/shared/projections/ContractStateProjection.kt]: Shall we add also a Profile to this one? in case we want to disable it?

File: src/test/java/com/hedvig/contract/shared/models/ContractRepositoryTest.kt (hunk 32 above)
Comment (meletios) [reply src/test/java/com/hedvig/contract/shared/models/ContractRepositoryTest.kt]: 🤯 oh yeah, we need to tests this

File: src/main/java/com/hedvig/contract/houses/models/HouseAgreementState.kt (hunk 8 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/houses/models/HouseAgreementState.kt]: Not for now, i guess? lets be something simple for now. Lets leave it as a ElementCollection

File: src/main/java/com/hedvig/contract/shared/models/objects/CoInsured.kt (hunk 16 above)
Comment (meletios) [reply src/main/java/com/hedvig/contract/shared/models/objects/CoInsured.kt]: lets leave it for now

File: src/main/java/com/hedvig/contract/apartments/aggregates/ApartmentAgreement.kt (hunk 3 above)
Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/apartments/aggregates/ApartmentAgreement.kt]: TODO: Remove

File: src/main/java/com/hedvig/contract/shared/stereotypes/ContractStereotype.kt (hunk 24 above)
Comment (vonElfvin) [reply src/main/java/com/hedvig/contract/shared/stereotypes/ContractStereotype.kt]: TODO: Remove if not used
